pkg/virt-operator/resource/apply/apps.go

```diff
@@ -157,12 +158,16 @@ func (r *Reconciler) patchDaemonSet(oldDs, newDs *appsv1.DaemonSet) (*appsv1.Dae
 
 func (r *Reconciler) getCanaryPods(daemonSet *appsv1.DaemonSet) []*corev1.Pod {
 	canaryPods := []*corev1.Pod{}
+	var hash string
+	if daemonSet.Annotations != nil {
+		hash = daemonSet.Annotations[v1.KubeVirtPodTemplateAnnotationHash]
```

Comment:
You should be able to simply use `daemonSet.Annotations[v1.KubeVirtPodTemplateAnnotationHash]`